### DISCUSSION TITLE
Best practice for logging the http request in an HTTP triggered Azure Function

### DISCUSSION THREAD

**Discussion by pim-simons** [arcus-azure/arcus.observability#175] [Q&A]
In an HTTP triggered Azure Function I want to log the http request, I also use a try catch containing the actual logic of the Azure Function.

My code looks like this (removed the parts that are not relevant):
```
try
{
	...
	return new ObjectResult(response);
}
catch (Exception ex)
{
	_logger.LogError(ex, "Exception occurred. " + ex.ToString());
	req.HttpContext.Response.StatusCode = 500;
	return (ActionResult)new ExceptionResult(ex, true);
}
finally
{
	_logger.LogRequest(req.HttpContext.Request, req.HttpContext.Response, stopWatch.Elapsed);
}
```

As you can see I log the request in the finally statement to make sure the request is logged even when an error occurs.
I manually set the HttpContext.Response.StatusCode to 500 to indicate an error.

I am just wondering if this is the right way to go, is there a better way to update the HttpContext.Response.StatusCode instead of manually assigning it?

**Comment by stijnmoreels** [comment]
There's indeed no (built-in) middleware to catch this, so you should do it yourself. There's is though an overload where you can pass along the response HTTP status code instead of the actual response where you set the status code manually.

```csharp
_logger.LogRequest(req.HttpContext.Request, 500, stopWatch.Elapsed);
```

I guess it's dependent on the situation. Adding @tomkerkhove to this.
